CenterForOpenScience/osf.io
# Remove backrefs for OSF-[5896, 5898, 5899, 5900, 5901, 5906, 5907, 5909]

Refs various, general parent is https://openscience.atlassian.net/browse/OSF-5674 

## Purpose
We are attempting to reduce the number of unused backrefs on our models, as backrefs create side effects (record locking) that are not ideal.

This PR is intended to encompass the low-hanging fruit: simple backrefs that can be easily replaced with small queries.

## Summary of changes
- Remove various unused backrefs (see diff)
- Replace usages of these backrefs with direct database queries

## Side effects/ testing notes
If all goes well, there will be no user-facing side effects, and OSF functionality will not change. Below is a list of features that could be affected by these changes.

- Make sure the list of active registrations appears correctly (eg for registrations tab on project pages)
- Watch (or unwatch) a project. Reload the page and make sure the number of watchers updates correctly. Make sure that you correctly receive updates about projects you are watching.
- Do something that would trigger a notification email (transactional or digest) on a project. Make sure that you receive the email correctly with no errors.
- Fork a project. Ensure the list of forks displays correctly, and the fork button updates with the correct fork count.
- Create a new project, using another project as a template. (fork button--> "copy project structure"). Verify that the "templated from" counter in the modal updates to show the correct count. 
- Node links should still work

There may be some issues with deletion as well. We don't really delete many things (users and nodes just get hidden via a flag), but try things like unwatching a node, and see if that gives errors.


## Subtasks
https://openscience.atlassian.net/browse/OSF-5896
https://openscience.atlassian.net/browse/OSF-5896
https://openscience.atlassian.net/browse/OSF-5898
https://openscience.atlassian.net/browse/OSF-5899
https://openscience.atlassian.net/browse/OSF-5900
https://openscience.atlassian.net/browse/OSF-5901
https://openscience.atlassian.net/browse/OSF-5906 (badges, unused)
https://openscience.atlassian.net/browse/OSF-5907 (badges, unused)
https://openscience.atlassian.net/browse/OSF-5909
